modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java

```diff
@@ -85,19 +79,14 @@ public LocalStackContainer withServices(Service... services) {
      * @return an {@link AwsClientBuilder.EndpointConfiguration}
      */
     public AwsClientBuilder.EndpointConfiguration getEndpointConfiguration(Service service) {
-
-        if (delegate == null) {
-            throw new IllegalStateException("LocalStack has not been started yet!");
-        }
-
-        final String address = delegate.getContainerIpAddress();
+        final String address = getContainerIpAddress();
         String ipAddress = address;
         try {
             ipAddress = InetAddress.getByName(address).getHostAddress();
         } catch (UnknownHostException ignored) {
 
         }
-        ipAddress = ipAddress + ".xip.io";
+        ipAddress = ipAddress + ".nip.io";
```

Comment:
Nitpick, but I assume this is not an IP address anymore after concatenating the String :stuck_out_tongue:

Reply:
I blame the original author ;)

Reply:
FYI I changed to nip.io because xip.io is unreliable (and caused some CI failures)